application/backend/app/services/dispatchers/mqtt.py

```diff
@@ -63,6 +66,20 @@ def __init__(
         self.client = mqtt_client or self._create_default_client()
         self._connect()
 
+    @staticmethod
+    def _get_credentials(auth_required: bool) -> tuple[str | None, str | None]:
```

Comment:
Why not keeping this in the model as a property? In theory, it can be reused in other places.